src/vs/workbench/contrib/terminal/browser/terminalStatusList.ts

```diff
@@ -94,7 +94,7 @@ export class TerminalStatusList extends Disposable implements ITerminalStatusLis
 			this._onDidRemoveStatus.fire(existingStatus);
 			this._statuses.delete(existingStatus.id);
 		}
-		if (!this._statuses.has(status.id)) {
+		if (!this._statuses.has(status.id) && (this.primary === undefined || status.icon)) {
 			const oldPrimary = this.primary;
 			this._statuses.set(status.id, status);
 			this._onDidAddStatus.fire(status);
```

Comment:
We still want to add the status here, just not change the primary status to one that does not have an icon. Statuses show as an icon but also in the hover.

Reply:
what's odd to me is `this.primary` isn't set anywhere

Reply:
and `oldPrimary` and `newPrimary` are both set to `his.primary`

Reply:
oh I see how this works... nvm